python/lsst/daf/butler/core/config.py

```diff
@@ -140,6 +148,66 @@ def extractFile(self, filename):
         return yaml.load(response["Body"], Loader)
 
 
+@dataclass
+class Resource:
```

Comment:
Is there a way to unify filesystem paths or even URIs with these `Resource` paths so that they appear to the rest of the code as a uniform class instead of having to type-case everywhere?

Reply:
Adding `pkg_resources://lsst.daf.butler/configs/datastore.yaml` URI support to ButlerURI might work but really needs `ButlerURI` to be updated to have concrete methods for checking existence and even opening a stream.  I think some of this ties in with @TallJimbo wondering if ButlerURI could be changed so that the constructor returns specialist S3ButlerURI, PosixFileURI and LocalFileURI implementations.